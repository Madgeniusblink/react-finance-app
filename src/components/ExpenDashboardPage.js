import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export { ExpenDashboardPage as default }