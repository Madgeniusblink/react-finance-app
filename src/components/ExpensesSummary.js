import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral'
import getVisibleExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'



export const ExpensesSummary = ({ expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpensessTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totaling {formattedExpensessTotal}</h1>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)