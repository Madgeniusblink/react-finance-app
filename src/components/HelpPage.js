import React from 'react'

const HelpPage = () => (
    <div>
        This is from my new HELP PAGE
    </div>
)

export { HelpPage as default }