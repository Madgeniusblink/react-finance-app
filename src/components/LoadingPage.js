import React from 'react'

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/Pacman-loader.gif" />
        <h1>Loading...</h1>
    </div>
)

export default LoadingPage