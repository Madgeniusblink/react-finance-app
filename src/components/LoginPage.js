import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export const LoginPage = ({ startLogin }) => (
    <div>
        <Link to="/dashboard">
            <button onClick={startLogin}>
                Login!
            </button>
        </Link>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)