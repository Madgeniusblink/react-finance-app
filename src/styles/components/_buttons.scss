.login-button {
    background: $dark-blue;
    color: $white;
    display: inline-block;
    border: none;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    border-radius: $s-size;
    -webkit-box-shadow: 0 8px 6px -6px black;
	   -moz-box-shadow: 0 8px 6px -6px black;
	        box-shadow: 0 8px 6px -6px black;
}

.login-button:hover {
    background-color: #4CAF50;
    color: $white;
}

.button {
    background: $blue;
    border: none;
    color: $white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--link:hover {
    color: red;
}

.button--secondary {
    background: #8888;
}
.button--secondary:hover {
    background: red;
}