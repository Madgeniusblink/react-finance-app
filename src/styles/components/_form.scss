.form {
    display: flex;
    flex-direction: column;
    >* {
        margin-bottom: $m-size;
    }
    @media (max-width: $desktop-breakpoint) {
        align-items: center;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}