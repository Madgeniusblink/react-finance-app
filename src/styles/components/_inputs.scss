.text-input {
    border: 1px solid #cacccd;
    height: 48px;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;

    @media (max-width: $desktop-breakpoint) {
        width: 316px;
    }

}

.select {
    @extend .text-input;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}