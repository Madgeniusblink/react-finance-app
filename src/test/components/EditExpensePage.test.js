import React from 'react'
import { shallow } from 'enzyme'
import { EditExpensePage } from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

let editExpense, history, removeExpense, wrapper;

beforeEach(() => {
    editExpense = jest.fn()
    history = { push: jest.fn() }
    removeExpense = jest.fn()
    wrapper = shallow(
    <EditExpensePage 
        editExpense={editExpense} 
        history={history} 
        removeExpense={removeExpense} 
        expense={expenses[2]}/>
    )
})

test('should render EditExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('should handle editExpense ', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[2])
})

test('should handle removeExpense ', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper).toMatchSnapshot()
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(removeExpense).toHaveBeenLastCalledWith({
        id: expenses[2].id
    })
})

