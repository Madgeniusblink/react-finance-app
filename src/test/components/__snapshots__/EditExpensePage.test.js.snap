// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle removeExpense  1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 39.98,
        "createdAt": 345600000,
        "description": "Credit Card",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 39.98,
        "createdAt": 345600000,
        "description": "Credit Card",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
